ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java

```diff
@@ -228,14 +232,24 @@ private void analyzeReplDump(ASTNode ast) throws SemanticException {
   }
 
   // REPL LOAD
-  private void initReplLoad(ASTNode ast) throws SemanticException {
+  private void initReplLoad(ASTNode ast) throws HiveException {
     path = PlanUtils.stripQuotes(ast.getChild(0).getText());
     int numChildren = ast.getChildCount();
     for (int i = 1; i < numChildren; i++) {
       ASTNode childNode = (ASTNode) ast.getChild(i);
       switch (childNode.getToken().getType()) {
         case TOK_DBNAME:
           dbNameOrPattern = PlanUtils.stripQuotes(childNode.getChild(0).getText());
+          for (String dbName : Utils.matchesDb(db, dbNameOrPattern)) {
+            Database database = db.getDatabase(dbName);
+            if (database != null) {
+              if (ReplChangeManager.isSourceOfReplication(database)) {
+                LOG.error("Cannot load database " + dbName +
+                        " as it is a source of replication");
+                throw new SemanticException(ErrorMsg.REPL_TARGET_IS_THE_SOURCE_OF_REPLICATION.getMsg());
```

Comment:
Why do we need to fail for this case? Isn't it enough to ensure, we don't overwrite this property when we load?

Reply:
remove this error code